### PR TITLE
[TreeView] Refactor the tree view internals to prepare for headless API

### PR DESCRIPTION
**Pull request by flaviendelangle**
- [x] Move `extractPluginParamsFromProps` inside `useTreeView`
- [x] Simplify the `LogExpandedItems` doc example (remove support for the `root` slot)
- [x] Remove the default plugin list in favor of one list per Tree View component we export (so that `RichTreeView` can have plugin that `SimpleTreeView` does not have)

### CODE REVIEW

File: packages/x-tree-view/src/internals/useTreeView/extractPluginParamsFromProps.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  const paramsLookup = {} as Record<keyof PluginParams, true>;
+  plugins.forEach((plugin) => {
+    Object.assign(paramsLookup, plugin.params);
+  });
+
+  const pluginParams = {} as PluginParams;
+  const forwardedProps = {} as Omit<TProps, keyof PluginParams>;
+
+  Object.keys(props).forEach((propName) => {
+    const prop = props[propName as keyof typeof props] as any;
```
Comment (alexfauquette) [inline packages/x-tree-view/src/internals/useTreeView/extractPluginParamsFromProps.ts:48]: We had similar issues for normalizing plugins in charts without using `any`. If one day you find a better solution than your current one, I would be interested :)

**Review by alexfauquette — COMMENTED** [review]
Seems good, but I don't know tree view well enough to spot room for improvement

File: packages/x-tree-view/src/internals/useTreeView/extractPluginParamsFromProps.ts (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-tree-view/src/internals/useTreeView/extractPluginParamsFromProps.ts:48]: image

Given the types of `Object.keys` I think there is no better solution for now :cry:

**Review by noraleonte — APPROVED** [review]
Looks good, nice improvement🚀 I really like that we now have a plugin list per component 💙

**Pull request merged**